cmd/soroban-rpc/internal/jsonrpc.go

```diff
@@ -53,16 +60,42 @@ func NewJSONRPCHandler(cfg *config.LocalConfig, params HandlerParams) Handler {
 		"getTransaction":      methods.NewGetTransactionHandler(params.TransactionStore),
 		"sendTransaction":     methods.NewSendTransactionHandler(params.Logger, params.TransactionStore, cfg.NetworkPassphrase, params.CoreClient),
 		"simulateTransaction": methods.NewSimulateTransactionHandler(params.Logger, params.LedgerEntryReader, params.PreflightGetter),
-	}, nil)
-	corsMiddleware := cors.New(cors.Options{
-		AllowedOrigins: []string{"*"},
-		AllowedHeaders: []string{"*"},
-		AllowedMethods: []string{"GET", "PUT", "POST", "PATCH", "DELETE", "HEAD", "OPTIONS"},
-	})
+	}, &bridgeOptions)
 
 	return Handler{
 		bridge:  bridge,
 		logger:  params.Logger,
-		Handler: corsMiddleware.Handler(bridge),
+		Handler: bridge,
+	}
+}
+
+type rpcLogger struct {
+	logger *log.Entry
+}
+
+func (r *rpcLogger) LogRequest(ctx context.Context, req *jrpc2.Request) {
+	r.logger.WithFields(log.F{
+		"subsys":   "jsonrpc",
+		"req":      middleware.GetReqID(ctx),
+		"json_req": req.ID(),
+		"method":   req.Method(),
+		// TODO: is this overkill?
+		"params": req.ParamString(),
+	}).Info("starting JSONRPC request")
+}
+
+func (r *rpcLogger) LogResponse(ctx context.Context, rsp *jrpc2.Response) {
+	// TODO: Print the elapsed time (there doesn't seem to be a way to it with with jrpc2, since
+	//       LogRequest cannot modify the context)
+	logger := r.logger.WithFields(log.F{
+		"subsys":   "jsonrpc",
+		"req":      middleware.GetReqID(ctx),
+		"json_req": rsp.ID(),
+		// TODO: is this overkill?
+		"result": rsp.ResultString(),
+	})
+	if err := rsp.Error(); err != nil {
+		logger = logger.WithField("error", err.Error())
 	}
+	logger.Info("finished JSONRPC request")
```

Comment:
this one would go away, right ?

Reply:
what do you mean? why would it go away?

Reply:
I think that you've added quite many log entries here. I think that it would work well for debugging, but might be overkill for production.
i.e. the amount of logging might be such that would end up impacting the server throughput.

I think that for now it's probably just fine, but for the long run, we'll need to update the logging to limit it to 1 log line per incoming request.